src/main/java/greencity/dto/specification/SpecificationNameDto.java

```diff
@@ -5,11 +5,13 @@
 import javax.validation.constraints.NotNull;
 import lombok.AllArgsConstructor;
 import lombok.Data;
+import lombok.EqualsAndHashCode;
 import lombok.NoArgsConstructor;
 
 @Data
 @NoArgsConstructor
 @AllArgsConstructor
+@EqualsAndHashCode(exclude = {"name"})
 public class SpecificationNameDto {
     @NotNull(message = ValidationConstants.EMPTY_SPECIFICATION_NAME)
     @NotEmpty(message = ValidationConstants.EMPTY_SPECIFICATION_NAME)
```

Comment:
@NotNull is inside @NotEmpty annotation, so you need not use @NotNull here. But also you should thing about @NotBlank annotation.